sys/hoon.hoon

```diff
@@ -4264,296 +4270,477 @@
-          ::  prefix: before each entry
-          ::  finale: after all entries
-          ::
-          =/  prefix  (trip sigil.u.indef.u.tall.plum)
-          =/  finale  (trip final.u.indef.u.tall.plum)
-          ::  if, no children, then, just prelude and finale
-          ::
-          ?:  =(~ blocks)
-            %+  weld
-              ?~(prelude ~ [0 prelude]~)
-            ?~(finale ~ [0 finale]~)
-          ::  if, no :prefix
-          ::
-          ?:  =(~ prefix)
-            ::  kids: flat list of child lines
-            ::  tab:  amount to indent kids
-            ::
-            =/  kids  `(list [indent=@ud text=tape])`(zing blocks)
-            =*  tab   =+((lent prelude) ?+(- 2 %0 0, %1 2, %2 4))
-            ::  indent kids by tab
-            ::
-            =.  kids  (turn kids |=([@ud tape] [(add tab +<-) +<+]))
-            ::  prepend or inject prelude
-            ::
-            =.  kids  
-              ?:  =(~ prelude)  kids
-              ::  if, no kids, or prelude doesn't fit
-              ::
-              ?:  |(?=(~ kids) (gte +((lent prelude)) indent.i.kids))
-                ::  don't inject, just add to head if needed
-                ::
-                [[0 prelude] kids]
-              ::  inject: prelude 
-              ::
-              =*  inject  %+  weld
-                            prelude
-                          %+  runt 
-                            [(sub indent.i.kids (lent prelude)) ' ']
-                          text.i.kids
-              [[0 inject] t.kids]
-            ::  append finale
-            ::
-            ?~  finale  kids
-            (weld kids ^+(kids [0 finale]~))
-          ::  else, with :prefix
-          ::
-          ::  append :finale 
-          ::
-          =-  ?~  finale  -
-              (weld - ^+(- [0 finale]~))
-          ^-  (list [indent=@ud text=tape])
-          ::  clear: clearance needed to miss prefix
-          ::
-          =/  clear  (add 2 (lent prefix))
-          %-  zing
-          ::  combine each subtree with the prefix
-          ::
-          %+  turn  blocks
-          |=  =(list [indent=@ud text=tape])
-          ^+  +<
-          ::  tab: depth to indent
-          ::
-          =*  tab  ?~(list 0 (sub clear (min clear indent.i.list)))
-          =.  list  (turn list |=([@ud tape] [(add tab +<-) +<+]))
-          ?~  list  ~
-          :_  t.list
-          :-  0
-          %+  weld  
-            prefix
-          (runt [(sub indent.i.list (lent prefix)) ' '] text.i.list)
+      ::  %tree: Try to render a text tree in tall mode.
+      ::
+      ::  We want to render this in tall mode. First, verify that there
+      ::  the plum has a tall render (if not, fall back to `linear`
+      ::  formatting), then render all the subplums, and then render
+      ::  them in one of three ways:
+      ::
+      ::  - If the `plumfmt` contains an `indef` and that indef has
+      ::    no prefix, then this is variable-arity rune with a terminator:
```

Comment:
this is variable-arity rune -> this is a variable-arity rune

Reply:
BAP style